irmaseal-pkg/src/middleware/metrics.rs

```diff
@@ -0,0 +1,120 @@
+use crate::server::POSTGUARD_CLIENTS;
+use crate::util::*;
+use actix_http::header::HeaderValue;
+use actix_web::{
+    body::MessageBody,
+    dev::{Service, ServiceRequest, ServiceResponse},
+};
+use futures::Future;
+
+pub(crate) fn collect_metrics<
+    B: MessageBody,
+    S: Service<ServiceRequest, Response = ServiceResponse<B>, Error = actix_web::Error>,
+>(
+    req: ServiceRequest,
+    srv: &S,
+) -> impl Future<Output = Result<ServiceResponse<B>, actix_web::Error>> {
+    if let Some(Ok(header)) = req.headers().get(PG_CLIENT_HEADER).map(HeaderValue::to_str) {
```

Comment:
[If-let chains](https://rust-lang.github.io/rfcs/2497-if-let-chains.html) would be nice here.